src/controller/BridgeGameController.js

```diff
@@ -0,0 +1,108 @@
+const BridgeGame = require("../BridgeGame");
+const OutputView = require("../view/OutputView");
+const InputView = require("../view/InputView");
+const { close } = require("../utils/utils");
+const BridgegLengthValidator = require("../utils/BridgeLengthValidator");
+const DirectionValidator = require("../utils/DirectionValidator");
+const RegameCommandValidator = require("../utils/RegameCommandValidator");
+const BridgeAnswerPath = require("../BridgeAnswerGenerator");
+const { STATUS, STATE } = require("../constants/message");
+const BridgeMap = require("../BridgeMap");
+
+class BridgeGameController {
+  constructor() {
+    this.model = {
+      bridgeGame: new BridgeGame(),
+      answerBridge: new BridgeAnswerPath(),
+      bridgeMap: new BridgeMap(),
+    };
+  }
+
+  updateBridgeLengthFromUser() {
+    return InputView.readBridgeSize(this.#updateAnswerBridgeAccordingInput.bind(this));
+  }
+
+  #updateAnswerBridgeAccordingInput(bridgeSize) {
+    this.#updateAnswerBridgeSize(bridgeSize);
+    this.#updateAnswerBridgePath();
+  }
+
+  #updateAnswerBridgeSize(bridgeSize) {
+    this.model.answerBridge.size = Number(bridgeSize);
+  }
+
+  #updateAnswerBridgePath() {
+    if (this.#isBridgeLengthNotValid()) return this.updateBridgeLengthFromUser();
+    this.model.answerBridge.buildBridge();
+    this.updateMoveDirectionFromUser();
+  }
+
+  #isBridgeLengthNotValid() {
+    return !BridgegLengthValidator.validate(this.model.answerBridge.size);
+  }
+
+  updateMoveDirectionFromUser() {
+    this.#isGameCleared()
+      ? this.#quit()
+      : InputView.readMoving(this.#updateUserMovement.bind(this));
+  }
+
+  #isGameCleared() {
+    return this.model.answerBridge.size === this.model.bridgeGame.getMyCurrentPosition();
+  }
+
+  #updateUserMovement(direction) {
+    if (this.#isDirectionNotValid(direction)) return this.updateMoveDirectionFromUser();
+    if (this.model.bridgeGame.move(direction, this.model.answerBridge.getbridge())) {
+      this.model.bridgeMap.updateMyPositionForward(direction, STATE.VALID.symbol);
+      return this.#showMovedPath().updateMoveDirectionFromUser();
+    }
+    this.model.bridgeMap.updateMyPositionForward(direction, STATE.NOT_VALID.symbol);
+    return this.#showMovedPath().#askUserRestart();
+  }
+
+  #isDirectionNotValid(direction) {
+    return !DirectionValidator.validate(direction);
+  }
+
+  #askUserRestart() {
+    InputView.readGameCommand(this.#updateRestartOrNot.bind(this));
+  }
+
+  #updateRestartOrNot(restartStatus) {
+    if (this.#isRegameCommandNotValid(restartStatus)) this.#askUserRestart();
+    if (restartStatus === STATUS.HOPE_RESTART) this.#regame();
+    if (restartStatus === STATUS.HOPE_QUIT) this.#quit();
```

Comment:
위 유효성 검사로 인해서 `STATUS.HOPE_RESTART` 값과 `STATUS.HOPE_QUIT` 값만 들어올 수 있어 보입니다! 🤓

`return` 을 이용해서 if문을 한 번만 하는게 좋아보이긴 한데 어떻게 생각하실까요? 🤔🤔 

```suggestion
    if (this.#isRegameCommandNotValid(restartStatus)) this.#askUserRestart();
    if (restartStatus === STATUS.HOPE_RESTART) return this.#regame();
    return this.#quit();
```

Reply:
else느낌이 나기에 굳이 if문으로 지정 해주었는데 확신이 있는 만큼 return문으로 한번 만 하는 것도 좋아 보이네요 ..ㅎㅎ 
좋은 의견 감사합니다..!